src/NServiceBus.Core.Tests/Recoverability/RecoverabilityExecutorTests.cs

```diff
@@ -1,22 +1,58 @@
 ﻿namespace NServiceBus.Core.Tests.Recoverability
 {
-    using NUnit.Framework;
-    using Testing;
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Pipeline;
+    using Extensibility;
+    using NServiceBus.Pipeline;
     using Transport;
+    using NUnit.Framework;
 
     [TestFixture]
-    public class DiscardRecoverabilityActionTests
+    public class RecoverabilityExecutorTests
     {
         [Test]
-        public void Discard_action_should_discard_message()
+        public async Task Should_use_error_context_extensions_as_extensions_root()
```

Comment:
Maybe we should split this test into 2 (upward and downward) and then we can also improve the naming.